sct-commons/src/main/java/org/lfenergy/compas/sct/commons/scl/ied/LDeviceAdapter.java

```diff
@@ -359,6 +377,74 @@ protected boolean hasControlBlockCreationCapability(ControlBlockEnum controlBloc
 
     private boolean hasCBNameConf(TServiceSettings tServiceSettings) {
         return tServiceSettings.isSetCbName()
-            && (TServiceSettingsNoDynEnum.CONF.equals(tServiceSettings.getCbName()));
+                && (TServiceSettingsNoDynEnum.CONF.equals(tServiceSettings.getCbName()));
+    }
+
+    /**
+     * Update and/or create Monitoring LNs (LSVS and LGOS) into LDSUIED from ExtRefs binding
+     *
+     * @param tExtRefs           ExtRefs for which source Control Blocks (Goose or SMV) should be monitored
+     * @param tSystemLNGroupEnum LNClass value for LN to monitor (Only LGOS and LSVS are taken into this enum)
+     * @return a list of SclReport Objects that contains errors
+     */
+    public List<SclReportItem> manageMonitoringLns(List<TExtRef> tExtRefs, String doName, TSystemLNGroupEnum tSystemLNGroupEnum) {
+        if (tExtRefs.isEmpty() || (!TSystemLNGroupEnum.LGOS.equals(tSystemLNGroupEnum)
+                && !TSystemLNGroupEnum.LSVS.equals(tSystemLNGroupEnum))) {
+            return Collections.emptyList();
+        }
+        List<SclReportItem> sclReportItems = new ArrayList<>();
+        getLNAdapters().stream().filter(lnAdapter -> tSystemLNGroupEnum.value().equals(lnAdapter.getLNClass())).findFirst()
+                .ifPresentOrElse(lnAdapter -> {
+                    ResumedDataTemplate filter = createDaiFilter(lnAdapter, doName);
+                    List<ResumedDataTemplate> foundDais = lnAdapter.getDAI(filter, true);
+                    if (foundDais.isEmpty()) {
+                        sclReportItems.add(SclReportItem.warning(lnAdapter.getXPath() + "/DOI@name=\"" + doName + "\"/DAI@name=\"setSrcRef\"/Val",
+                                "The DAI cannot be updated"));
+                    } else {
+                        ResumedDataTemplate daToUpdate = foundDais.get(0);
+                        TLN lnToUpdate = lnAdapter.getCurrentElem();
+                        int i = 1;
+                        for (TExtRef tExtRef : tExtRefs) {
+                            if (i != 1) {
+                                this.getCurrentElem().getLN().add(lnToUpdate);
+                            }
+                            updateNewCreatedLnDaiValue(lnToUpdate, tExtRef, String.valueOf(i), daToUpdate).ifPresent(sclReportItems::add);
+                            i++;
+                            lnToUpdate = copySclElement(lnAdapter.getCurrentElem(), TLN.class);
+                        }
+                    }
+                }, () -> sclReportItems.add(SclReportItem.warning(getXPath(), "There is no LN %s present in LDevice".formatted(tSystemLNGroupEnum.value()))));
+        return sclReportItems;
```

Comment:
The method can only return 0 or 1 SclReportItem. You could replace its return value type by `Optional<SclReportItem>`.